core/base/inc/TDirectory.h

```diff
@@ -296,7 +301,7 @@ can be replaced with the simpler and exception safe:
    static void         DecodeNameCycle(const char *namecycle, char *name, Short_t &cycle, const size_t namesize = 0);
    static void         EncodeNameCycle(char *buffer, const char *name, Short_t cycle);
 
-   ClassDefOverride(TDirectory,5)  //Describe directory structure in memory
+   ClassDefOverride(TDirectory,6)  //Describe directory structure in memory
```

Comment:
Note, that `TDirectory` only use manual schema evolution (because by definition they are read before the `StreamerInfo`.   Changing the onfile format of TDirectory is "onerous" as we have to think about forward compatibility and propagate the change to other tools ( `uproot`, `root in go`, etc.)

Reply:
For the record, the `CanOwn/AcceptsRegisters` property should be 'transient', otherwise different `TFile` would act differently depending on how they were created (so it should not be recorded in fBits).

Another issue to consider is whether this is a file level or directory level property (as implemented here, it is a 'directory' level property and and in particular (as implemented so far :) ) it does not apply to sub-directory of the `TFile`.